### PR TITLE
Add getCount

### LINKED ISSUE #81: feat: Derive the count of a `limit`ed dare.get request
This ticket supersedes #31.

Obtain the number of rows a count request would have retrieved had it not been limited.

e.g.

```js
// Request with a limit
const dataset = dare.get(request);

// How many rows matched to generate pagination
const count = dare.getCount(request);
```

**Approach**

Create a new function **`dare.getCount(request)`**. This will be able to take an existing request object in the first parameter and retrieve the number of records which match this query.

This can be done by stripping away any `fields`, `orderby`, `limit` and `start` props.
Defining a single field containing `COUNT(DISTINCT ${groupby})` where groupby is any custom or derived `groupby` applied to the matching dataset.

### PR DESCRIPTION
**Pull request by MrSwitch**
Fixes #81

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -410,6 +410,51 @@ The following statement includes all results from the main table, but does not a
...
+|----------|-------------------|----------------
+| table    | string            | Name of the table to access
+| filter   | Hash (key=>Value) | Query Object
+| options  | Hash (key=>Value) | Additional Options
+
+e.g.
+
+```javascript
+dare.getCount('profile', {first_name: 'Andrew'});
+// SELECT COUNT(DISTINCT a.id) FROM profile WHERE name = 'Andrew' LIMIT 1;
```
Comment (ikhemissi) [inline README.md]: `COUNT(DISTINCT id)`: a.id will be unknown

File: README.md [hunk 2]
```diff
@@ -410,6 +410,51 @@ The following statement includes all results from the main table, but does not a
...
+const requestOptions = {
+	table: 'profile',
+	filter: {
+		first_name: 'Andrew'
+	},
+	limit: 10
+};
+
+// Get the first 10 items, and the number of possible rows
+const [items, foundRows] = Promise.all([
```
Comment (ikhemissi) [inline README.md]: `await Promise.all`

File: package.json [hunk 3]
```diff
@@ -41,22 +43,27 @@
...
   },
   "dependencies": {},
+  "publishConfig": {
+    "access": "public",
+    "prerelease": true,
+    "tag": "next"
+  },
   "release": {
-    "branch": "master",
+    "branch": "next",
```
Comment (ikhemissi) [inline package.json]: was `next` intentionally left here ? I thought it was needed just for testing

File: src/index.js [hunk 4]
```diff
@@ -143,6 +143,58 @@ Dare.prototype.get = async function get(table, fields, filter, opts = {}) {
 
 };
 
+/**
+ * GetCount
+ *
+ * @param {string} table - Name of the table to query
+ * @param {object} filter - Filter Object to query
+ * @param {object} opts - An Options object containing all other request options
+ * @returns {integer} Number of matched items
```
Comment (ikhemissi) [inline src/index.js]: @returns {Promise<number>}

File: package.json (hunk 3 above)
Comment (MrSwitch) [reply package.json]: Just removed

**Pull request merged**